lib/block/header.ex

```diff
@@ -3,56 +3,76 @@ defmodule Block.Header do
-    receipts_root: @empty_trie,      # He TRIE({∀i < kBRk, i ∈ P : p(i, LR(BR[i]))})
-    logs_bloom: <<0::2048>>,         # Hb bloom
-    difficulty: nil,                 # Hd
-    number: nil,                     # Hi
-    gas_limit: 0,                    # Hl
-    gas_used: 0,                     # Hg
-    timestamp: nil,                  # Hs
-    extra_data: <<>>,                # Hx
-    mix_hash: nil,                   # Hm
-    nonce: nil,                      # Hn
-  ]
+  @empty_trie MerklePatriciaTree.Trie.empty_trie_root_hash()
+  @empty_keccak [] |> ExRLP.encode() |> :keccakf1600.sha3_256()
+
+            # Hp P(BH)Hr
+  defstruct parent_hash: nil,
+            # Ho KEC(RLP(L∗H(BU)))
+            ommers_hash: @empty_keccak,
+            # Hc
+            beneficiary: nil,
+            # Hr TRIE(LS(Π(σ, B)))
+            state_root: @empty_trie,
+            # Ht TRIE({∀i < kBTk, i ∈ P : p(i, LT (BT[i]))})
+            transactions_root: @empty_trie,
+            # He TRIE({∀i < kBRk, i ∈ P : p(i, LR(BR[i]))})
+            receipts_root: @empty_trie,
+            # Hb bloom
+            logs_bloom: <<0::2048>>,
+            # Hd
+            difficulty: nil,
+            # Hi
+            number: nil,
+            # Hl
+            gas_limit: 0,
+            # Hg
+            gas_used: 0,
+            # Hs
+            timestamp: nil,
+            # Hx
+            extra_data: <<>>,
+            # Hm
+            mix_hash: nil,
+            # Hn
+            nonce: nil
 
   # As defined in Eq.(35)
   @type t :: %__MODULE__{
-    parent_hash: EVM.hash,
-    ommers_hash: EVM.trie_root,
-    beneficiary: EVM.address,
-    state_root: EVM.trie_root,
-    transactions_root: EVM.trie_root,
-    receipts_root: EVM.trie_root,
-    logs_bloom: binary(), # TODO
-    difficulty: integer() | nil,
-    number: integer() | nil,
-    gas_limit: EVM.val,
-    gas_used: EVM.val,
-    timestamp: EVM.timestamp | nil,
-    extra_data: binary(),
-    mix_hash: EVM.hash | nil,
-    nonce: <<_::64>> | nil, # TODO: 64-bit hash?
-  }
+          parent_hash: EVM.hash(),
+          ommers_hash: EVM.trie_root(),
+          beneficiary: EVM.address(),
+          state_root: EVM.trie_root(),
+          transactions_root: EVM.trie_root(),
+          receipts_root: EVM.trie_root(),
+          # TODO
+          logs_bloom: binary(),
+          difficulty: integer() | nil,
+          number: integer() | nil,
+          gas_limit: EVM.val(),
+          gas_used: EVM.val(),
+          timestamp: EVM.timestamp() | nil,
+          extra_data: binary(),
+          mix_hash: EVM.hash() | nil,
+          # TODO: 64-bit hash?
+          nonce: <<_::64>> | nil
+        }
 
   # The start of the Homestead block, as defined in Eq.(13) of the Yellow Paper (N_H)
   @homestead_block 1_150_000
 
-  @initial_difficulty 131_072 # d_0 from Eq.(40)
-  @minimum_difficulty @initial_difficulty # Mimics d_0 in Eq.(39), but variable on different chains
+  # d_0 from Eq.(40)
```

Comment:
These comments (I think) made more sense inline

Reply:
@hayesgm yes, I agree inline comments look better and they are more readable. But elixir formatter does not think so :). It does not have many configuration parameters. I think it was made that way to get rid of code style debates. 

It even formats inline comments not equivalently. I went over all changed files checking and fixing comments but unfortunately missed some broken comments. I will fix them.